### ISSUE TITLE
Train Contrastive STS model

### ISSUE THREAD

**Issue by juanmc2005** [juanmc2005/SimilarityLearning#12]
Determine a threshold to build positive and negative STS pairs and train a model with contrastive loss.
Validate with Spearman score (should discuss which is the best method).

**Comment by juanmc2005** [comment]
Training is possible with cosine distance. Euclidean distance is giving NaN weights after a couple of optimizer steps. I'll have to fix that later.

**Comment by juanmc2005** [comment]
Reopening as there are other hyperparameters and approaches to evaluate

**Comment by juanmc2005** [comment]
After last meeting with @sophierosset , we are considering this experiment finished.
A significance test will be done later as suggested by @saharghannay and @sophierosset
